Documentation/user-guides/configuring-cluster-monitoring.md

```diff
@@ -1,9 +1,161 @@
+Despite this, to avoid users from missconfiguring the monitoring stack of their clusters not all configuration parameters are exposed.
+
+The ConfigMap `cluster-monitoring-config` in the `openshift-monitoring` namespace is the root for all configuration of the core infrastruture monitoring. Here we can configure the following:
+
+```yaml
+# Holds configuration for everything Alertmanager related in the cluster
+# from configuring nodeSelectors to enabeling UserAlertManagerConfig
+alertmanagerMain: <struct>
+# Boolean flag to enable UserWorkloadMonitoring
+enableUserWorkload: <bool>
+# Holds configuration related with prometheus-adapter, nodeSelector and tolerations
+# and Audit level
+k8sPrometheusAdapter: <struct>
+# Holds configuration related with kubeStateMetrics, nodeSelector and tolerations
+kubeStateMetrics: <struct>
+# Holds configuration related with the prometheus pods that will be managed by
+# the prometheus-operator from configuring nodeSelector to RemoteWrite
+prometheusK8s: <struct>
+# Holds configuration related with prometheus-operator, nodeSelector, tolerations 
+# and logLevel 
+prometheusOperator: <struct>
+# Holds configuration related with openshift-state-metrics, nodeSelector and tolerations
+openshiftStateMetrics: <struct>
+# Holds configuration related with thanos-querier, nodeSelector, tolerations 
+thanosQuerier: <struct>
+```
+
+### alertmanagerMain
+
+Holds configuration for all things alermanager
+
+```yaml
+alertmanagerMain:
+  # Boolean flag to enable or disable the alertmanager instance that 
+  # will live under openshift-monitoring 
+  enabled: <bool>
+  # Boolean flag to enable or disable the alertmanager instance for 
+  # user workload monitoring  under openshift-user-workload-monitoring
+  enableUserAlertmanagerConfig: <bool>
+  # Log level to be used by the main alertmanager instance, possible values are
+  # info, warn, debug. #TODO validate this
+  logLevel: <string>
+  # Define which Nodes the Pods are scheduled on.
+  nodeSelector: <struct>
+  # Defines the alertmanager Pods tolerations.
+  tolerations: <struct>
+  # Define resources requests and limits for alertmanager Pods.
+  resources: <struct>
+  # Allows configuring persistent storage for the alertmanager Pods 
+  # users can configure storageClass and size of volume.
+  volumeClaimTemplate: <struct>
+```
+
+Example
+
+```yaml
+alertmanagerMain:
+  enabled: true
+  enableUserAlertmanagerConfig: true
+  logLevel: "debug"
+  nodeSelector:
+    kubernetes.io/os: linux
+  tolerations:
+    - operator: Exists
+  resources:
+    requests:
+      cpu: 5m
+      memory: 30Mi
+```
+
+### enableUserWorkloadMonitoring
+
+Enabels or disabels user workload monitoring, by default is `false`. When enabeled, CMO will create a new namespace `openshift-user-workload-monitoring` to where it will deploy the following components:
```

Comment:
nit: Enables or disables by default disabled

When enabled CMO will create a new namespace  called `openshift-user-workload-monitoring` to which it will deploy the following components